app/models/illiad_openurl.rb

```diff
@@ -0,0 +1,35 @@
+###
+#  Class to handle creation of ILLiad OpenURL request
+###
+class IlliadOpenurl
+  def initialize(scan, redirect_url)
+    @scan = scan
+    @redirect_url = redirect_url
+  end
+
+  def illiad_page
+    { 'Action': '10', 'Form': '30', 'rft.genre': 'scananddeliver' }
+  end
+
+  def illiad_item_data
+    { 'rft.location': @scan.origin,
+      'rft.call_number': @scan.holdings.first.callnumber,
+      'rft.item': @scan.holdings.first.barcode }
+  end
+
+  def illiad_citation
+    { 'rft.jtitle': @scan.item_title,
+      'rft.au': @scan.data[:authors],
+      'rft.pages': @scan.data[:page_range],
+      'rft.atitle': @scan.data[:section_title] }
+  end
+
+  def illiad_callback
+    { 'scan_referrer': @redirect_url }
+  end
+
+  def to_url
+    illiad_qy = [illiad_page, illiad_item_data, illiad_citation, illiad_callback].reduce({}, :merge)
```

Comment:
`illiad_qy` looks like it should be its own method (`query_params` or something)?